BZ1825250: Managing cluster resources (#39793)
openshift/openshift-docs: modules/cluster-resources.adoc

[id="support-cluster-resources_{context}"]
= Interacting with your cluster resources

> You can interact with cluster resources using OpenShift CLI (`oc`) in {product-title}. The cluster resources that you see after running `oc api-resources` command are editable.

Reviewer: ```suggestion
You can interact with cluster resources by using the OpenShift CLI (`oc`) tool in {product-title}. The cluster resources that you see after running the `oc api-resources` command can be edited.
```

Author: acked

.Prerequisites

> * Access to the cluster as a user with the `cluster-admin` role.
> * Access to the web console or the `oc` installed.

Reviewer: These are sentence fragments and thus wouldn't need periods. But generally, I prefer making them complete sentences (your call):
```suggestion
* You have access to the cluster as a user with the `cluster-admin` role.
* You have access to the web console or you have installed the `oc` CLI tool.
```

Author: acked

.Procedure

> . To see what configuration operators applied, run the following command:

Reviewer: Capitalize Operators ([ref link](url))
```suggestion
. To see which configuration Operators have been applied, run the following command:
```

Author: acked

> . To see which configuration operators have been applied, run the following command:

Reviewer: Is this "Operators" in the [OpenShift](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#operator-capitalization) context?
```suggestion
. To see which configuration Operators have been applied, run the following command:
```

Author: Yes. In the Openshift context

Author: acked

+
[source,terminal]
----
$ oc api-resources -o name | grep config.openshift.io
----

> . To know what cluster resources you can configure, run the following command:

Reviewer: For consistency, I suggest using "To see" in each command example.
```suggestion
. To see what cluster resources you can configure, run the following command:
```

Author: acked

+
[source,terminal]
----

> $ oc explain ${resource_name}.config.openshift.io

Reviewer 2: Hi @xenolinux! Just a quick note that if you're noting something that the user replaces, we use the format `<resource_name>`. (As opposed to ${resource_name}). Not urgent, but if you wouldn't mind following up with a separate PR to adjust them, that would be appreciated!

Author: Thanks for noticing it @bergerhoffer. I opened a PR https://github.com/openshift/openshift-docs/pull/40670. Can you please take a look and merge if looks good?

Author: I believe QE ack won't be required for this minor change.

Reviewer 2: Agreed on no QE needed, thanks @xenolinux, looking!

----

> . To know configuration of CRD objects in the cluster, run the following command:

Reviewer: ```suggestion
. To see the configuration of custom resource definition (CRD) objects in the cluster, run the following command:
```

Author: acked

+
[source,terminal]
----

> $ oc get $config -o yaml

Reviewer 3: command here is not correct . should be : oc get apiservers.config -o yaml . or `oc get ${resource_name}.config -o yaml`

Author: Acked

> . To see the configuration of custom resource definition CRD objects in the cluster, run the following command:

Reviewer: ```suggestion
. To see the configuration of custom resource definition (CRD) objects in the cluster, run the following command:
```

Author: acked

----

> . To edit the cluster resource configuration, run the following commands:

Reviewer: It's just one command, so singular:
```suggestion
. To edit the cluster resource configuration, run the following command:
```

Author: acked

+
[source,terminal]
----

> $ oc edit $config -o yaml

Reviewer 3: Command here is not correct . should be  : `oc edit apiservers.config -o yaml`  .

Author: acked
